app/controllers/incomeSources/add/BusinessReportingMethodController.scala

```diff
@@ -0,0 +1,198 @@
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc._
+import services.{BusinessReportingMethodService, IncomeSourceDetailsService}
+import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException}
+import views.html.errorPages.CustomNotFoundError
+import views.html.incomeSources.add.BusinessReportingMethod
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class BusinessReportingMethodController @Inject()(val authenticate: AuthenticationPredicate,
+                                                  val authorisedFunctions: FrontendAuthorisedFunctions,
+                                                  val checkSessionTimeout: SessionTimeoutPredicate,
+                                                  val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                  val retrieveBtaNavBar: NavBarPredicate,
+                                                  val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                                  val retrieveNino: NinoPredicate,
+                                                  val view: BusinessReportingMethod,
+                                                  val businessReportingMethodService: BusinessReportingMethodService,
+                                                  val customNotFoundErrorView: CustomNotFoundError)
+                                                 (implicit val appConfig: FrontendAppConfig,
+                                                  mcc: MessagesControllerComponents,
+                                                  val ec: ExecutionContext,
+                                                  val itvcErrorHandler: ItvcErrorHandler,
+                                                  val itvcErrorHandlerAgent: AgentItvcErrorHandler
+                                                 )
+  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  private def handleRequest(isAgent: Boolean, id: String)
+                           (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
+
+    val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessReportingMethodController.submitAgent(id) else
+      controllers.incomeSources.add.routes.BusinessReportingMethodController.submit(id)
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+    val redirectUrl: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAddedController.showAgent(id) else
+      controllers.incomeSources.add.routes.BusinessAddedController.show(id)
+
+    if (incomeSourcesEnabled) {
+      businessReportingMethodService.checkITSAStatusCurrentYear.flatMap {
+        case true =>
```

Comment:
checkITSAStatusCurrentYear - its unclear from the name what the boolean result means.
may I suggest: isITSAStatusEligible and then the true/false values make more sense